src/mender-update/daemon/states.cpp

```diff
@@ -391,13 +391,24 @@ void UpdateDownloadState::DoDownload(Context &ctx, sm::EventPoster<StateEvent> &
 	}
 	ctx.deployment.artifact_payload.reset(new artifact::Payload(std::move(exp_payload.value())));
 
-	auto handler = [&poster](error::Error err) {
+	auto handler = [&poster, &ctx](error::Error err) {
 		if (err != error::NoError) {
 			log::Error(err.String());
 			poster.PostEvent(StateEvent::Failure);
 			return;
 		}
 
+		auto exp_payload = ctx.deployment.artifact_parser->Next();
+		if (exp_payload) {
+			log::Error("Multiple payloads are not yet supported in daemon mode.");
+			poster.PostEvent(StateEvent::Failure);
```

Comment:
Q: Does it not have to return? This will post a `Failure` and then a `Success` right after

Reply:
Ooh, good catch! And rookie mistake by me! 😆